resource/ui/ts/models/timeseries.ts

```diff
@@ -43,68 +44,183 @@ module Models {
-                end_nanos: end.getTime() * 1.0e6,
-                queries: series.map((r) => {return {
-                    name: r,
-                    aggregator: agg,
-                };}),
+        export interface QueryRequestSet {
+            start_nanos:number;
+            end_nanos:number;
+            queries:QueryRequest[];
+        }
+
+
+        /**
+         * select contains selectors for use in time series queries. A database
+         * query contains one or more selectors which define the time series
+         * that are queried and how the data from each series is aggregated.
+         */
+        export module select {
+            /**
+             * Selector is an interface for reading the data in a Selector.
+             */
+            export interface Selector {
+                /**
+                 * request returns a QueryRequest object based on this selector.
+                 */
+                request():QueryRequest;
+                /**
+                 * title returns a display-friendly title for this series.
+                 */
+                title():string;
+            }
+
+            /**
+             * AvgSelector selects the average value of the supplied time series.
+             */
+            class AvgSelector {
+                constructor(private series_name:string) {}
+
+                title = Utils.chainProp(this, this.series_name);
+
+                request = ():QueryRequest => {
+                    return {
+                        name:this.series_name,
+                        aggregator:QueryAggregator.AVG,
+                    }
+                }
             }
-        
-            return m.request({url:url, method:"POST", extract:nonJsonErrors, data:data})
-                .then((d:QueryResultSet) => {
-                    // Populate missing collection fields with empty arrays.
-                    if (!d.results) {
-                        d.results = [];
+
+            /**
+             * AvgRateSelector selects the rate of change of the average value
+             * of the supplied time series.
+             */
+            class AvgRateSelector {
+                constructor(private series_name:string) {}
+
+                title = Utils.chainProp(this, this.series_name);
+
+                request = ():QueryRequest => {
+                    return {
+                        name:this.series_name,
+                        aggregator:QueryAggregator.AVG_RATE,
                     }
-                    d.results.forEach((r) => {
-                        if (!r.datapoints) {
-                            r.datapoints = []
-                        }
-                    });
-                    return d;
-                });
+                }
+            }
+
+            /**
+             * Avg instantiates a new AvgSelector for the supplied time series.
+             */
+            export function Avg(series:string):AvgSelector {
```

Comment:
Yes, for fluent programming.  You can do this with the function:
```
var s = select.Avg("series").title("My series");
```

Without the function, it would have to be:

```
var s = (new AvgSelector("series")).title("My series");
```